docs/en/index.md

```diff
@@ -159,39 +159,265 @@ Nested GridFields
+
+As an example, here we can use a typical hierarchy of the `Group` model, where another `Group` can serve as a parent.
+
 ```php
-class NestedObject extends DataObject
+namespace App\Extensions;
+
+use SilverStripe\Core\Extension;
+use SilverStripe\Security\Group;
+use Symbiote\GridFieldExtensions\GridFieldNestedForm;
+
+class MySecurityAdminExtension extends Extension
 {
-	private static $has_one = [
-		'Parent' => ParentObject::class
-	];
-
-	public function getNestedConfig(): GridFieldConfig
-	{
-		$config = new GridFieldConfig_RecordViewer();
-		return $config;
-	}
+    public function updateGridFieldConfig($config)
+    {
+        if ($this->owner->getModelClass() === Group::class) {
+            $config->addComponent(GridFieldNestedForm::create());
+        }
+    }
 }
 ```
 
-You can also modify the default config (a `GridFieldConfig_RecordEditor`) via an extension to the nested model class, by implementing
-`updateNestedConfig`, which will get the config object as the first parameter.
+Or also view a list of all members of a given group. Notice we call `setRelationName()` to explicitly state the relation which should be displayed in the Nested GridField.
+
 ```php
-class NestedObjectExtension extends DataExtension
+namespace App\Extensions;
+
+use SilverStripe\Core\Extension;
+use SilverStripe\Security\Group;
+use Symbiote\GridFieldExtensions\GridFieldNestedForm;
+
+class MySecurityAdminExtension extends Extension
 {
-	public function updateNestedConfig(GridFieldConfig &$config)
-	{
-		$config->removeComponentsByType(GridFieldPaginator::class);
-	}
+    public function updateGridFieldConfig($config)
+    {
+        if ($this->owner->getModelClass() === Group::class) {
+            $config->addComponent(GridFieldNestedForm::create()->setRelationName('Members'));
+        }
+    }
+}
+```
+
+```yml
+SilverStripe\Admin\SecurityAdmin:
+extensions:
+    - App\Extensions\MySecurityAdminExtension
+```
+
+Another way to use Nested GridField together with `DataObjects` that do not have the `Hierarchy` extension but have `has_many` relationships with other objects.
+Let's say you have the following `DataObject` that has multiple levels of relationships, and an admin section where the data of this object will be displayed.
+You want the user to be able to view information regarding this object and all nested objects on the same page, without the need to navigate to each object individually.
+In this case, you can use the following approach.
+
+```php
+namespace App\Models;
+
+use SilverStripe\ORM\DataObject;
+
+class ParentNode extends DataObject
+{
+    //...
+
+    private static $has_many = [
+        'ChildNodes' => BranchNode::class,
+    ];
+}
+```
+
+You can define your own custom GridField config for the nested GridField configuration by implementing a `getNestedConfig` on your nested model. Notice this method should return a [`GridFieldConfig`](api:SilverStripe\Forms\GridField\GridFieldConfig) object.
```

Comment:
```suggestion
You can define your own custom GridField config for the nested GridField configuration by implementing a `getNestedConfig` on your nested model. Notice this method should return a `GridFieldConfig` object.
```
These docs aren't pulled into docs.silverstripe.org so we can't use that syntax unfortunately.

Reply:
Updated